vm/src/builtins/classmethod.rs

```diff
@@ -53,22 +53,34 @@ impl GetDescriptor for PyClassMethod {
         cls: Option<PyObjectRef>,
         vm: &VirtualMachine,
     ) -> PyResult {
-        let (zelf, obj) = Self::_unwrap(zelf, obj, vm)?;
-        let cls = cls.unwrap_or_else(|| obj.class().clone().into());
-        let callable = zelf.callable.lock().clone();
-        Ok(PyBoundMethod::new_ref(cls, callable, &vm.ctx).into())
+        let (zelf, _obj) = Self::_unwrap(zelf, obj, vm)?;
+        let cls = cls.unwrap_or_else(|| _obj.class().clone().into());
+        let call_descr_get: PyResult<PyObjectRef> = zelf.callable.lock().get_attr("__get__", vm);
+        match call_descr_get {
+            Err(_) => Ok(PyBoundMethod::new_ref(cls, zelf.callable.lock().clone(), &vm.ctx).into()),
+            Ok(call_descr_get) => vm.invoke(&call_descr_get, (cls.clone(), cls)),
+        }
     }
 }
 
 impl Constructor for PyClassMethod {
     type Args = PyObjectRef;
 
     fn py_new(cls: PyTypeRef, callable: Self::Args, vm: &VirtualMachine) -> PyResult {
-        PyClassMethod {
+        let doc = callable.get_attr("__doc__", vm);
+
+        let result = PyClassMethod {
             callable: PyMutex::new(callable),
         }
-        .into_ref_with_type(vm, cls)
-        .map(Into::into)
+        .into_ref_with_type(vm, cls)?;
+        let obj = PyObjectRef::from(result);
+
+        match doc {
+            Err(_) => None,
+            Ok(doc) => Some(obj.set_attr("__doc__", doc, vm)),
+        };
```

Comment:
```suggestion
        if let Ok(doc) = doc {
            obj.set_attr("__doc__", doc, vm)?;
        }
```
set_attr result need to be checked with `?`

Reply:
Good points!